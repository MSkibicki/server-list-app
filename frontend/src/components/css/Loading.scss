@import "./Global.scss";

.loading {
  text-align: center;
  margin-top: 10em;
  color: $nav-color;
  text-shadow: 2px 2px 2px $black;
}
