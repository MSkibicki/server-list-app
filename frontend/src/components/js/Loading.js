import React from "react";
import "../css/Loading.scss";

const Loading = () => {
  return (
    <div className="loading">
      <h1 className="loading-text">Loading...</h1>
    </div>
  );
};

export default Loading;
