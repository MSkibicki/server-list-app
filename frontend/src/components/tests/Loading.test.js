import Loading from "./../js/Loading";

describe("<Loading />", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<Loading />);

    expect(wrapper).toMatchSnapshot();
  });

  it("displays h1 text correctly", () => {
    const wrapper = shallow(<Loading />);

    const text = wrapper.find(".loading-text").text();

    expect(text).toEqual("Loading...");
  });

  it("has correct div classname", () => {
    const wrapper = shallow(<Loading />);

    const divClass = wrapper.find("div").hasClass("loading");

    expect(divClass).toBe(true);
  });

  it("has correct h1 classname", () => {
    const wrapper = shallow(<Loading />);

    const headingClass = wrapper.find("h1").hasClass("loading-text");

    expect(headingClass).toBe(true);
  });
});
