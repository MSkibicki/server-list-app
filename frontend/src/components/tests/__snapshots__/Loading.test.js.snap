// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> renders correctly 1`] = `
<div
  className="loading"
>
  <h1
    className="loading-text"
  >
    Loading...
  </h1>
</div>
`;
